::std::strchr return type differs from ::std::string position arg

in src/main.cpp, line 728:

```c++
auto* pos = ::std::strchr(desc, '=');
```

a few lines later:
```c++
auto name = ::std::string(desc, pos);
```

compiler throws an error:
```
src/main.cpp:728:52: error: Invalid conversion from 'char*' to 'std::basic_string<char>::size_type {aka long unsigned int}'
```
Changing pos to *pos makes it compile here, but not sure if that is the correct solution.

--- a/src/main.cpp
+++ b/src/main.cpp
@@ -773,7 +773,7 @@ ProgramParams::ProgramParams(int argc, char *argv[])
                     exit(1);
                 }
                 const char* desc = argv[++i];
-                auto* pos = ::std::strchr(desc, '=');
+                const char* pos = ::std::strchr(desc, '=');
                 if( pos == nullptr ) {
                     ::std::cerr << "--extern takes an argument of the format name=path" << ::std::endl;
                     exit(1);

